Assets/Dependencies/SmashBrew/Character/States/MovementState.cs

```diff
@@ -187,14 +187,26 @@ struct MovementInfo {
             bool success = (!IsCrounching && JumpCount > 0 && JumpCount <= MaxJumpCount
                 && (Input.GetKeyDown(KeyCode.W) || Input.GetKeyDown(KeyCode.UpArrow)));
             if (success) {
-                CurrentLedge = null;
+                if (CurrentLedge != null) {
+                    LeaveLedge();
+                }
                 OnJump.SafeInvoke();
                 PhysicsState.SetVerticalVelocity(_jumpPower[MaxJumpCount - JumpCount]);
                 CmdJump();
             }
             return success;
         }
 
+        void LeaveLedge() {
+            if (CurrentLedge != null) {
```

Comment:
Only runs if the CurrentLedge is not null. Change it to: 
```csharp
if (CurrentLedge == null)
   return;
```
Prevents the pyramid of doom.